Library/Homebrew/utils.rb

```diff
@@ -770,3 +770,24 @@ def number_readable(number)
   (numstr.size - 3).step(1, -3) { |i| numstr.insert(i, ",") }
   numstr
 end
+
+def truncate_text_to_approximate_size(s, max_bytes, options = {})
+  front_weight = options[:front_weight] || 0.5
+  glue = options[:glue] || "\n[...snip...]\n"
+  return s if s.bytesize <= max_bytes
+
+  if front_weight < 0 || front_weight > 1
+    raise "opts[:front_weight] must be between 0.0 and 1.0"
+  end
+  scale = 0.99 # just a tad small to allow for the glue
+  while true
```

Comment:
(And when I say, "UTF-8 is a variable-length encoding", that's a simplification: this implementation works for all encodings, not just UTF-8. But it has to be done this way because some possible input encodings, including UTF-8 which is what we use pretty much everywhere, are variable-width. I believe this implementation is not encoding-dependent.)